librad/src/git/storage/pool.rs

```diff
@@ -3,30 +3,53 @@
 // This file is part of radicle-link, distributed under the GPLv3 with Radicle
 // Linking Exception. For full terms see the included LICENSE file.
 
-use std::sync::{Arc, Mutex};
+use std::{
+    ops::Deref,
+    sync::{Arc, Mutex},
+};
 
-use deadpool::managed::{Manager, RecycleResult};
+use deadpool::managed::{self, Manager, Object, RecycleResult};
 
 use super::{Error, Storage};
 use crate::{paths::Paths, signer::Signer};
 
 pub type Pool = deadpool::managed::Pool<Storage, Error>;
-pub type Pooled = deadpool::managed::Object<Storage, Error>;
+pub type PoolError = managed::PoolError<Error>;
 
-/// Wrapper so we can use [`Pooled`] as `AsRef<Storage>`.
+#[async_trait]
+pub trait Pooled {
+    async fn get(&self) -> Result<PooledRef, PoolError>;
+}
+
+#[async_trait]
+impl Pooled for Pool {
+    async fn get(&self) -> Result<PooledRef, PoolError> {
+        self.get().await.map(PooledRef::from)
+    }
+}
+
+/// A reference to a pooled [`Storage`].
 // TODO: may go away once https://github.com/bikeshedder/deadpool/pull/69
```

Comment:
fyi: it looks like this is in `0.7.0` https://github.com/bikeshedder/deadpool/blob/deadpool-v0.7.0/src/managed/mod.rs#L152-L162